cluster-api/cloud/google/clients/computeservice.go

```diff
@@ -0,0 +1,71 @@
+package clients
+
+import (
+	compute "google.golang.org/api/compute/v1"
+	"net/http"
+	"net/url"
+)
+
+type ComputeService interface {
+	ImagesGet(project string, image string) (*compute.Image, error)
+	ImagesGetFromFamily(project string, family string) (*compute.Image, error)
+	InstancesDelete(project string, zone string, targetInstance string) (*compute.Operation, error)
+	InstancesGet(project string, zone string, instance string) (*compute.Instance, error)
+	InstancesInsert(project string, zone string, instance *compute.Instance) (*compute.Operation, error)
+	ZoneOperationsGet(project string, zone string, operation string) (*compute.Operation, error)
+}
+
+type computeServiceImpl struct {
+	service *compute.Service
+}
+
+func NewComputeService(client *http.Client) (ComputeService, error) {
```

Comment:
You should delete the `ComputeService` interface from this package, change `computeServiceImpl` to `ComputeService`, and return a pointer to it here.